main.ts

```diff
@@ -150,27 +148,45 @@ const textSliderTemplate = (textSlide: ITextSlide) => `
    </div>
 </section>`;
 
-
-// should extend abstract class BaseSlider
-class TextSlider {
-
- // TODO: finish the method addSlides
- // addSlides(slide: TextSlide[]) {
- // }
-
- // use textSliderTemplate
+class TextSlider extends BaseSlider {
+  constructor(selector: string, showButtons = true) {
+    super(selector, showButtons);
+  }
+
+  addSlides(slides: ITextSlide[]) {
+    if (Array.isArray(images)) {
+      const sliderTemplate = (source: ITextSlide) => {
+        const imgString: string = textSliderTemplate(source);
+        return createDOMElFromString(imgString);
+      };
+
+      this.slides = slides.map(source => sliderTemplate(source));
+    }
+  }
 }
 
-
-class AutomaticSlider  {
- // TODO: automatic slider should be the descendant of the SimpleSlider or
- // TextSlider but with automatic slideshow and without buttons
-
- // add next() to the render method
+class AutomaticSlider extends TextSlider {
+  constructor(selector: string, showButtons = false) {
+    super(selector, showButtons);
+  }
+
+  public render(): void {
+    super.render();
+    let t = this;
+    setInterval(function tick() {
+      t.next();
```

Comment:
I didn't get why 
```
setInterval(this.next, 2000) 
```
is not working. Any ideas? The same for 
```
setInterval(function tick() {
       this.next();
}, 200); 
```
But I believe that in the second example **this** represents another context. Anyway it's not clear for me with first example

Reply:
Ok, it seems that I missed something
just tried again
setInterval(this.next, 2000); 
and it works

Reply:
:-) js is awesome.

1. you can use arrow syntax 
setInterval(() => { this.next(); }, 200)

or 

2. "next" method could be a "class field"

next = () => {

}, 
so this will work setInterval(this.next, 200);

3. Or "bind it"

this.bindedNext = this.next.bind(this);

so this will work setInterval(this.bindedNext, 200);


p.s. your solution is the most old school way) and it works as well.